### ISSUE TITLE
feat: prompt mutation rate — control how much GEPA rewrites the prompt each iteration

### ISSUE THREAD

**Issue by LakshyAAAgrawal** [gepa-ai/gepa#277]
## Motivation

Users new to GEPA are often worried that the optimizer will completely rewrite their carefully crafted prompt in a single step — losing all the domain-specific wording, formatting, and constraints they already have. This is the **learning rate problem**: large updates are faster but riskier; small updates are safer but slower.

Currently GEPA's reflection LM can propose any change to the prompt — from a single-word tweak to a complete rewrite. There is no built-in way to constrain the magnitude of each mutation.

## Proposal

Add a `mutation_rate` parameter to `ReflectionConfig` (and the `gepa.optimize` API) that controls **how much of the prompt can change per iteration**, expressed as a fraction of the current prompt length:

```python
result = gepa.optimize(
    seed_candidate={"system_prompt": "Your carefully crafted prompt..."},
    ...
    mutation_rate=0.2,   # allow up to ~20% of the prompt to change per step
)
```

Or via `ReflectionConfig`:

```python
from gepa.optimize_anything import GEPAConfig, ReflectionConfig

config = GEPAConfig(
    reflection=ReflectionConfig(
        reflection_lm="openai/gpt-5",
        mutation_rate=0.2,   # 0.0 = no change, 1.0 = full rewrite (default)
    )
)
```

### How it would work

1. **Prompt-level instruction** (simplest, no parsing needed): when `mutation_rate < 1.0`, append an instruction to the reflection prompt template like:
   > *"Make targeted edits only. Your proposed prompt must retain at least 80% of the current prompt's content. Do not rewrite from scratch."*

2. **Post-hoc enforcement** (more robust): after the LM proposes a new prompt, compute the similarity to the old prompt (e.g. character-level diff ratio or token overlap). If the change exceeds the allowed rate, either:
   - Reject the proposal and ask for a smaller edit (retry with a stricter instruction)
   - Trim the diff to only apply the first N% of changes

### Suggested values

| `mutation_rate` | Behaviour |
|----------------|-----------|
| `1.0` (default) | Current behaviour — LM can rewrite entirely |
| `0.3` | Conservative — only ~30% of the prompt can change per step |
| `0.1` | Very conservative — small targeted edits only |
| `0.0` | No change allowed (effectively freezes the prompt) |

## Why This Matters

- **Trust**: users who have invested time in a prompt want confidence that GEPA will refine it, not replace it
- **Stability**: for production prompts, small incremental changes are easier to review, audit, and roll back
- **Convergence**: a lower mutation rate acts like a smaller learning rate — slower but more stable optimization trajectory

This is analogous to the `mutation_rate` / `step_size` parameter in evolutionary algorithms and the learning rate in gradient descent.
